### PR TITLE
[podsync] Build updated image

### LINKED ISSUE #83: [podsync] Build updated image
# Details

mxpv/podsync:v2.4
No Helm chart exists

**Describe the solution you'd like:**
Problems with existing images:
- original image has not been updated for a year. youtube-dl version is too old. Alpine version is old.
- It doesn't support parametrized config file placement. Can be overcome by: https://github.com/Marx2/homelab/blob/440421d9dd71219b826ae42974db600145ff42c3/cluster/media/podsync/podsync.yaml#L73

Original repository: https://github.com/mxpv/podsync
Newer fork with up-to-date ARM support: https://github.com/tuxpeople/docker-podsync

### PR DESCRIPTION
**Pull request by Marx2**
**Description of the change**

Build podsync image with newest youtube-dl
Play nice with Kubernetes deployment

- fixes #83

### CODE REVIEW

File: apps/podsync/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# hadolint ignore=DL3008
+RUN \
+  apt-get -qq update \
+  && \
+  apt-get -qq install -y --no-install-recommends \
+    ca-certificates \
+    ffmpeg \
+    tzdata \
+    ffmpeg \
+    python3-pip \
```
Comment (onedr0p) [inline apps/podsync/Dockerfile:47]: You don't need python3-pip, check this out

https://github.com/wernight/docker-youtube-dl/blob/6f39bcc1119edb0fa1a21f58f0e12b1005f72be9/Dockerfile#L16

all you need is python3

File: apps/podsync/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,71 @@
+FROM golang:1.16 as builder
+
```
Comment (onedr0p) [inline apps/podsync/Dockerfile:2]: You'll need to set the targetplatform and version args here too or else the won't be used in this stage.

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: apps/podsync/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  git clone https://github.com/mxpv/podsync.git . \
+  && git checkout ${VERSION} \
+  && go build -o /bin/podsync ./cmd/podsync
+
+
+# hadolint ignore=DL3007
+FROM ghcr.io/k8s-at-home/ubuntu:latest
+
+ARG TARGETPLATFORM
+ARG VERSION
```
Comment (onedr0p) [inline apps/podsync/Dockerfile:28]: You don't need these in this part of the stage

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: apps/podsync/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,71 @@
+FROM golang:1.16 as builder
+
+ENV CGO_ENABLED=0 \
+    GOPATH=/go \
+    GOBIN=/go/bin \
+    GO111MODULE=off
```
Comment (onedr0p) [inline apps/podsync/Dockerfile:6]: Probably need this set to on

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: apps/podsync/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+RUN \
+  export GOOS \
+  && GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) \
+  && export GOARCH \
+  && GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) \
+  && export GOARM \
+  && GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3 | cut -c2-) \
+  git clone https://github.com/mxpv/podsync.git . \
+  && git checkout ${VERSION} \
+  && go build -o /bin/podsync ./cmd/podsync
```
Comment (james-d-elliott) [inline apps/podsync/Dockerfile:21]:
```suggestion
  && git checkout ${VERSION} \
  && go mod download \
  && go build -o /bin/podsync ./cmd/podsync
```

**Review by james-d-elliott — CHANGES_REQUESTED** [review]

File: apps/podsync/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+RUN \
+  apt-get -qq update \
+  && \
+  apt-get -qq install -y --no-install-recommends \
+    ca-certificates \
+    ffmpeg \
+    tzdata \
+    ffmpeg \
+    python3-pip \
+  && \
```
Comment (onedr0p) [inline apps/podsync/Dockerfile:48]: You should also symlink python3 to python

e.g. `ln -s /usr/bin/python3 /usr/bin/python`

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: apps/podsync/Dockerfile (hunk 1 above)
Comment (Marx2) [reply apps/podsync/Dockerfile:47]: I've changed method of installing youtube-dl from apt package to pip package (python package delivers freshiest version). Without pip image building fails:
`#11 38.18 /bin/sh: 1: pip3: not found`

File: apps/podsync/Dockerfile (hunk 6 above)
Comment (Marx2) [reply apps/podsync/Dockerfile:48]: I'll add it, but why do I you think I need it? I don't call python directly

File: apps/podsync/Dockerfile (hunk 4 above)
Comment (Marx2) [reply apps/podsync/Dockerfile:6]: this was crucial change to make libraries working

File: apps/podsync/Dockerfile (hunk 5 above)
Comment (Marx2) [reply apps/podsync/Dockerfile:24]: seems not to be needed (after enabling GO111MODULE=on)

File: apps/podsync/PLATFORM [hunk 7]
```diff
@@ -0,0 +1 @@
+linux/amd64,linux/arm64
```
Comment (bjw-s) [inline apps/podsync/PLATFORM:1]: Missing newline

File: apps/podsync/VERSION [hunk 8]
```diff
@@ -0,0 +1 @@
+v2.4.0
```
Comment (bjw-s) [inline apps/podsync/VERSION:1]: Missing newline

**Review by bjw-s — CHANGES_REQUESTED** [review]

File: apps/podsync/entrypoint.sh [hunk 9]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+#shellcheck disable=SC1091
+source "/shim/umask.sh"
+source "/shim/vpn.sh"
+
+exec /app/podsync ${EXTRA_ARGS}
```
Comment (bjw-s) [inline apps/podsync/entrypoint.sh:7]: Missing newline

**Review by bjw-s — CHANGES_REQUESTED** [review]

File: apps/podsync/Dockerfile (hunk 1 above)
Comment (onedr0p) [reply apps/podsync/Dockerfile:48]: That's not what I'm asking...

Comment (onedr0p) [reply apps/podsync/Dockerfile:49]: Take a look at that Dockerfile I linked, that is how you should be installing ytdl.

File: apps/podsync/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+  apt-get -qq update \
+  && \
+  apt-get -qq install -y --no-install-recommends \
+    ca-certificates \
+    ffmpeg \
+    tzdata \
+    ffmpeg \
+    python3 \
+    wget \
+  && wget -O /usr/bin/youtube-dl https://github.com/ytdl-org/youtube-dl/releases/latest/download/youtube-dl \
```
Comment (onedr0p) [inline apps/podsync/Dockerfile:51]: Use curl, not wget

**Pull request merged**